Laravel-whib/routes/api.php

```diff
@@ -24,4 +24,14 @@
 Route::post('/deletefriend', [FriendsController::class, 'DeleteFriend'])->middleware('auth:sanctum');
 Route::get('/getusersbyname', [UserAuthController::class, 'GetUsersByName']);
 Route::get('/getuserbyid', [UserAuthController::class, 'GetUserById']);
-Route::get('/getuserfriends', [FriendsController::class, 'GetUserFriends'])->middleware('auth:sanctum');
\ No newline at end of file
+Route::get('/getuserfriends', [FriendsController::class, 'GetUserFriends'])->middleware('auth:sanctum');
+Route::get('/get-map-pins', [MapController::class, 'getMapPins'])->middleware('auth:sanctum');
+
+// Map pins routing
+Route::middleware(['auth:sanctum'])->group(function () {
+    Route::get('/map-pins', [MapController::class, 'index']);
+    Route::get('/map-pins/{mapPin}', [MapController::class, 'show']);
```

Comment:
Does this actually work? Like you get that pin from id or???

Reply:
This route definition will automatically bind the {mapPin} route parameter to the corresponding MapPin model instance, allowing you to use it as a parameter in the show method. With this setup, you can access the map pin ID directly within the $mapPin parameter in your controller method, and Laravel will take care of retrieving the model from the database.